mutation_order_gibbs.py

```diff
@@ -360,3 +425,60 @@ def _get_denom_update(self, old_denominator, prev_feat_idxs, feat_mut_step):
             prev_exp_theta_sum = self.exp_theta_sum[prev_feat_idxs] if len(prev_feat_idxs) else self.exp_theta_num_cols
             new_denom = old_denominator - prev_exp_theta_sum - np.sum(old_feat_exp_theta_sums) + np.sum(new_feat_exp_theta_sums)
         return float(new_denom)
+
+    def _get_risk_update(self, base_risk_vec, prev_mut_pos, feat_mut_step, new_denom):
```

Comment:
it would be really helpful for me to see the math written out in a tex file so that i can confirm the math makes sense here. I kind of understand residuals but not confident enough to check your code.